extensions/banner/deployment/src/main/java/io/quarkus/banner/deployment/BannerProcessor.java

```diff
@@ -0,0 +1,29 @@
+package io.quarkus.banner.deployment;
+
+import io.quarkus.banner.runtime.BannerConfig;
+import io.quarkus.banner.runtime.BannerRecorder;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.ExecutionTime;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.ConsoleFormatterBannerBuildItem;
+import io.quarkus.deployment.builditem.HotDeploymentWatchedFileBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.NativeImageResourceBuildItem;
+
+public class BannerProcessor {
+
+    @BuildStep
+    @Record(ExecutionTime.RUNTIME_INIT)
+    public ConsoleFormatterBannerBuildItem recordBanner(BannerRecorder recorder, BannerConfig config) {
+        return new ConsoleFormatterBannerBuildItem(recorder.provideBannerSupplier(config));
+    }
+
+    @BuildStep
+    HotDeploymentWatchedFileBuildItem watchBannerChanges(BannerConfig config) {
+        return new HotDeploymentWatchedFileBuildItem(config.path);
+    }
+
+    @BuildStep
+    NativeImageResourceBuildItem nativeImageResourceBuildItem(BannerConfig config) {
+        return new NativeImageResourceBuildItem(config.path);
+    }
```

Comment:
if we move the code that read the banner file to build time, we shouldn't need this build step.

Reply:
Very good point